x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/DeprecationRoleDescriptorConsumer.java

```diff
@@ -0,0 +1,146 @@
+                                        k -> new HashSet<String>());
+                                privilegesByIndex.addAll(Arrays.asList(indicesPrivilege.getPrivileges()));
+                            }
+                        }
+                    }
+                }
+                // sort by alias and by index for tests
+                final Map<String, Automaton> indexAutomatonMap = new HashMap<>();
+                for (final Map.Entry<String, Set<String>> privilegesByAlias : privilegesByAliasMap.entrySet()) {
+                    final String aliasName = privilegesByAlias.getKey();
+                    final Set<String> aliasPrivileges = privilegesByAliasMap.get(aliasName);
+                    final Automaton aliasPrivilegeAutomaton = IndexPrivilege.get(aliasPrivileges).getAutomaton();
+                    final TreeSet<String> inferiorIndexNames = new TreeSet<>();
+                    // check if the alias grants superiors privileges to the indices it points to
+                    for (final IndexMetaData indexMetadata : aliasOrIndexMap.get(aliasName).getIndices()) {
+                        final String indexName = indexMetadata.getIndex().getName();
+                        final Set<String> indexPrivileges = privilegesByIndexMap.get(indexName);
+                        // non null if this index matches the permission 
+                        if (indexPrivileges == null) {
+                            inferiorIndexNames.add(indexName);
+                        } else {
+                            final Automaton indexPrivilegeAutomaton = indexAutomatonMap.computeIfAbsent(indexName,
+                                    i -> IndexPrivilege.get(indexPrivileges).getAutomaton());
+                            if (false == Operations.subsetOf(indexPrivilegeAutomaton, aliasPrivilegeAutomaton)) {
+                                inferiorIndexNames.add(indexName);
+                            }
+                        }
+                    }
+                    if (false == inferiorIndexNames.isEmpty()) {
+                        final String logMessage = String.format(Locale.ROOT, ROLE_PERMISSION_DEPRECATION_STANZA, roleDescriptor.getName(),
+                                aliasName, String.join(", ", inferiorIndexNames));
+                        deprecationLogger.deprecated(logMessage);
+                    }
+                }
+                // mark role as checked for "today"
+                addToCache(cacheKey);
+            }
+        }
+    }
+
+    private boolean isCached(String cacheKey) {
+        return cacheKeys.contains(cacheKey);
+    }
+
+    private boolean addToCache(String cacheKey) {
+        return cacheKeys.add(cacheKey);
+    }
+
+    // package-private for testing
+    static String buildCacheKey(String roleName) {
+        final String daysSinceEpoch = Long.toString(ZonedDateTime.now(ZoneId.of("UTC")).toEpochSecond() / 86400000L);
```

Comment:
> Why all this trouble to convert it to a long-string rather than just using the ISO date?

No excuse 😞 It is now `ZonedDateTime.now(ZoneId.of("UTC")).format(DateTimeFormatter.BASIC_ISO_DATE)`